multi-checker/script/utils.py

```diff
@@ -53,15 +53,20 @@ def fixEncodingRecursively(startDir: str, ext: str):
     for p in files:
         fixEncoding(p)
 
-def findFile(p: str, sourceDir: str) -> Optional[str]:
+def findFile(p: str, sourceDir: str, ignoreCase: bool=False) -> Optional[str]:
     cand = pjoin(sourceDir, p)
     if isFile(cand):
         return cand
     for x in ls(sourceDir):
+        if ignoreCase and basename(x).lower() == p.lower() and isFile(x):
+            return x
         if isDir(x):
             cand = pjoin(x, p)
             if isFile(cand):
                 return cand
+            for y in ls(x):
+                if ignoreCase and basename(y).lower() == p.lower() and isFile(y):
```

Comment:
Where is the logic where we descend into the directory tree until we reach a directory that contains more than one subdirectory? There seems to be some duplicated logic.

Reply:
That's contained in the findSolutionDir method in the utils.py file. We can probably remove the block inside of the `if isDir(x):` block. What do you think?

Reply:
Yes, I think this should work.  The right way to find out is to ensure that a test for this feature exists. If it still runs after removing the block, everything is fine ;-)

Reply:
There is a test for this behavior (see `{pythonTestDir}/labortest_2/solution/`) and it fails after removing the block.


I'm not sure why we would want to support spreading solutions across multiple directories:
```
solution
├── aufgabe_01.py
└── subdir
    ├── aufgabe_01.py
    ├── aufgabe_02.py
    └── aufgabe_03.py
```

Reply:
This generality is not needed, indeed. So remove it!

Reply:
Done